src/Message/LtiMessage.php

```diff
@@ -77,7 +77,11 @@ public static function fromServerRequest(ServerRequestInterface $request): LtiMe
 
     public function toUrl(): string
     {
-        return sprintf('%s?%s', $this->url, http_build_query(array_filter($this->getParameters()->all())));
+        $separator = false !== strpos($this->url, '?')
+            ? '&'
+            : '?';
+
+        return sprintf('%s%s%s', $this->url, $separator, http_build_query(array_filter($this->getParameters()->all())));
```

Comment:
Since the library defines the query-parameter separator on the code level, I would suggest supplying `$arg_separator` to `http_build_query()`. Otherwise it may be overridden by `arg_separator.output` ini option.

```suggestion
        return sprintf('%s%s%s', $this->url, $separator, http_build_query(array_filter($this->getParameters()->all()), '', '&'));
```